src/main/java/gregtech/common/covers/detector/CoverDetectorEnergyAdvanced.java

```diff
@@ -88,72 +108,101 @@ private void compareValue(long value, long maxValue, long minValue) {
         }
     }
 
+    /**
+     * Compares the ratio value to min and max, and sets the output value accordingly.
+     * <p>
+     * Behaves like an SR Latch.
+     */
+    private void compareValue(float value, long maxValue, long minValue) {
+        if (value >= maxValue) {
+            this.outputAmount = inverted ? 15 : 0;
+        } else if (value <= minValue) {
+            this.outputAmount = inverted ? 0 : 15;
+        }
+    }
```

Comment:
I think this construction was recently implemented somewhere so it should be reused

Reply:
Tech had already asked me this, here's my response: https://github.com/GregTechCEu/GregTech/pull/1511#discussion_r1110581809.